src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,316 @@
+import datetime
+import xml.etree.ElementTree as ET
+import os
+import zipfile
+import glob
+import numpy as np
+
+
+
+class annotation:
+    # A parent class for annotation reader for Calibrarion, Noise, and poteltially L1
+    def __init__(self,path_annotation:str,polarization='VH',subswath=1):
+        #Attributes
+        self._IPF_version=None
+        self.path_annotation=path_annotation #Accepted format: SAFE .zip file, SAFE directory, or annotation .xml file
+        self.polarization=polarization
+        self.subswath=subswath #Subswath of the annotation file to load
+        self._xml_loaded=None
+        self.kind=''
+
+        #Data access point for the annotation .xml file
+        self.xml_et=None
+
+    def _get_search_phrase(self):
+        str_safe=os.path.basename(self.path_annotation).replace('.zip','')
+        token_safe=str_safe.split('_')
+
+        phrase_base='{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
```

Comment:
You probably want to validate `token_safe` to make sure that it has 3 elements? Otherwise, you raise an exception.

Reply:
Thanks @gshiroma for the detailed comments and suggestions. The code have gone through Pylint to make it more conform to PEP8.
The followings are my response to the comments you've posted.

I've added couple of line to raise an ValueError when `token_safe` does not look have enough elements

https://github.com/seongsujeong/s1-reader/blob/7e85dd7cda8d2ff3d36779894b9e0c52365d6149/src/s1reader/s1_annotation.py#L36-L37
```python
if len(token_safe)<3:
     raise ValueError(f'Invalid path_annotation: {self.path_annotation}')
```